src/honeypot.rs

```diff
@@ -177,7 +182,105 @@ impl<M: Middleware + 'static> HoneypotFilter<M> {
         }
 
         for (_, token) in tokens.iter().enumerate() {
-            self.validate_token(*token, None, None).await;
+            self.validate_token_on_simple_transfer(*token, None).await;
+        }
+    }
+
+    pub async fn validate_token_on_simulate_swap(
+        &mut self,
+        token_addr: H160,
+        pool_addr: H160,
+        buy_tax_criteria: Option<U256>,
+        sell_tax_criteria: Option<U256>,
+    ) {
+        let buy_tax_criteria = buy_tax_criteria.unwrap_or(U256::from(5));
+        let sell_tax_criteria = sell_tax_criteria.unwrap_or(U256::from(5));
+
+        self.simulator.deploy_simulator();
+
+        // seed the simulator with some safe token balance
+        let safe_token = self.safe_tokens.weth;
+
+        let amount_in_u32 = 1u32;
+        self.simulator.set_token_balance(
+            self.simulator.simulator_address,
+            safe_token,
+            WETH_DECIMALS,
+            WETH_BALANCE_SLOT,
+            amount_in_u32,
+        );
+
+        // buy with 0.1WETH
+        let weth_amount_in = U256::from(10i64.pow(WETH_DECIMALS.sub(1).into()));
+
+        // Buy Test
+        let buy_output = self.simulator.v2_simulate_swap(
+            weth_amount_in,
+            pool_addr,
+            safe_token,
+            token_addr,
+            true,
+        );
+        // (TargetedAmountOut, RealAfterBalance)
+        // out.0 is TargetedAmountOut that is calculated using amountIn in arg
+        // out.1 is the amount that is actually transfered to the simulator eoa address
+        let out = match buy_output {
+            Ok(out) => out,
```

Comment:
可読性のために`out.0`や`out.1`が何を扱っているかわかりやすくしたいです

```suggestion
        let (expected_amount_out, actual_amount_out) = match buy_output {
            Ok(out) => (expected_amount_out, actual_amount_out),
```